### PR TITLE
ci: add clippy

### PR DESCRIPTION
**Pull request by Roznovjak**
## Description
Resolve all clippy warnings and add clippy to github workflow

### CODE REVIEW

File: pallets/lbp/src/benchmarking.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ benchmarks! {
 		ensure!(PoolData::<T>::contains_key(&pool_id), "Pool does not exist.");
 
 		let pool_data = LBP::<T>::pool_data(&pool_id);
-		assert_eq!(pool_data.paused, false);
+		assert!(!pool_data.paused);
```
Comment (lumir-mrkva) [inline pallets/lbp/src/benchmarking.rs:230]: i don't like double negation 🙃

Comment (lumir-mrkva) [reply pallets/lbp/src/benchmarking.rs:230]: and i think that `assert_eq` is more appropriate in this case anyway

File: pallets/transaction-multi-payment/benchmarking/src/mock.rs [hunk 2]
```diff
@@ -163,11 +163,9 @@ impl AssetPairAccountIdFor<AssetId, u64> for AssetPairAccountIdTest {
 		let mut a = asset_a as u128;
 		let mut b = asset_b as u128;
 		if a > b {
-			let tmp = a;
-			a = b;
-			b = tmp;
+			std::mem::swap(&mut a, &mut b)
```
Comment (lumir-mrkva) [inline pallets/transaction-multi-payment/benchmarking/src/mock.rs:166]: fancy

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -582,7 +584,7 @@ parameter_types! {
 	// Bond for candidacy into governance
 	pub const CandidacyBond: Balance = 10_000 * DOLLARS;
 	// 1 storage item created, key size is 32 bytes, value size is 16+16.
-	pub const VotingBondBase: Balance = 1 * DOLLARS;
+	pub const VotingBondBase: Balance = DOLLARS;
```
Comment (lumir-mrkva) [inline runtime/src/lib.rs:587]: - [ ] we should really rename DOLLARS already, for example to UNIT

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]

File: pallets/lbp/src/benchmarking.rs (hunk 1 above)
Comment (Roznovjak) [reply pallets/lbp/src/benchmarking.rs:230]: There is only one negation :) I agree with Clippy (Clippy suggests to use assert macro). Using `assert_eq` is like using `if condition == true` anti-pattern.

File: pallets/transaction-multi-payment/benchmarking/src/mock.rs (hunk 2 above)
Comment (Roznovjak) [reply pallets/transaction-multi-payment/benchmarking/src/mock.rs:166]: ...standard. I would say it is fancy to perform swap on your own.

File: pallets/lbp/src/benchmarking.rs (hunk 1 above)
Comment (lumir-mrkva) [reply pallets/lbp/src/benchmarking.rs:230]: ok, you're right, but the second `!` is hard to spot, I actually had to read it multiple times what you';re trying to check, before it was much more obvious

Comment (lumir-mrkva) [reply pallets/lbp/src/benchmarking.rs:230]: i am not looking at it as predicate, but simply as value check - that's why actually asset_eq makes more sense to me

**Review by lumir-mrkva — APPROVED** [review]

**Pull request merged**